app/assets/javascripts/discourse/app/controllers/edit-topic-timer.js

```diff
@@ -20,53 +20,68 @@ export default Controller.extend(ModalFunctionality, {
   loading: false,
   isPublic: "true",
 
-  @discourseComputed("model.closed")
-  publicTimerTypes(closed) {
-    let types = [
-      {
-        id: CLOSE_STATUS_TYPE,
-        name: I18n.t(
-          closed ? "topic.temp_open.title" : "topic.auto_close.title"
-        ),
-      },
-    ];
+  @discourseComputed("model.closed", "model.category", "model.isPrivateMessage")
+  publicTimerTypes(closed, category, isPrivateMessage) {
+    let types = [];
 
     if (!closed) {
+      types.push({
+        id: CLOSE_STATUS_TYPE,
+        name: I18n.t("topic.auto_close.title"),
+      });
       types.push({
         id: CLOSE_AFTER_LAST_POST_STATUS_TYPE,
         name: I18n.t("topic.auto_close_after_last_post.title"),
       });
     }
 
-    types.push(
-      {
+    if (closed) {
+      types.push({
+        id: OPEN_STATUS_TYPE,
+        name: I18n.t("topic.auto_reopen.title"),
+      });
+    }
+
+    if (this.currentUser.staff) {
+      types.push({
+        id: DELETE_STATUS_TYPE,
+        name: I18n.t("topic.auto_delete.title"),
+      });
+    }
+
+    types.push({
+      id: BUMP_TYPE,
+      name: I18n.t("topic.auto_bump.title"),
+    });
+
+    if (this.currentUser.staff) {
+      types.push({
+        id: DELETE_REPLIES_TYPE,
+        name: I18n.t("topic.auto_delete_replies.title"),
+      });
+    }
+
+    if (closed) {
+      types.push({
+        id: CLOSE_STATUS_TYPE,
+        name: I18n.t("topic.temp_open.title"),
+      });
+    }
+
+    if (!closed) {
+      types.push({
         id: OPEN_STATUS_TYPE,
-        name: I18n.t(
-          closed ? "topic.auto_reopen.title" : "topic.temp_close.title"
-        ),
-      },
-      {
+        name: I18n.t("topic.temp_close.title"),
+      });
+    }
+
+    if ((category && category.read_restricted) || isPrivateMessage) {
```

Comment:
I think if a topic is "not visible" it can be a candidate for publishing. lets add that as well

Reply:
Yep, done